feat(tree): Allows annotated AllowedTypes to be added to schemas (#24478)
microsoft/FluidFramework: .changeset/wise-loops-tell.md

---
"@fluidframework/tree": minor
"__section": tree
---

> Allows users of alpha schema APIs to add metadata to AllowedTypes

Reviewer: Same issue as https://github.com/microsoft/FluidFramework/pull/24478/files/63f7ccc47d21ad5940bfd30fc2fe70d70d37a20e#r2069401208 .

```suggestion
Users of alpha schema APIs can now add metadata to AllowedTypes
```

> This change allows metadata to be added to AllowedTypes as well as individual types in a set of AllowedTypes.

Reviewer: references to type names should be code formatted, and the first reference should be linked.

```suggestion
This change allows metadata to be added to [`AllowedTypes`](https://fluidframework.com/docs/api/fluid-framework/allowedtypes-typealias) as well as individual types in a set of `AllowedTypes`.
```

> Users can define custom metadata by putting their AllowedTypes in an object with `metadata` and `types` properties:

Reviewer: ```suggestion
Users can define custom metadata by putting their `AllowedTypes` in an object with `metadata` and `types` properties:
```

```typescript

> schemaFactoryAlpha.arrayAlpha({
> 	metadata: {
> 		custom: "these allowed types are annotated",
> 	},
> 	types: [schemaFactory.string, schemaFactory.number],

Reviewer: ```suggestion
schemaFactoryAlpha.arrayAlpha({
	metadata: {
		custom: "these allowed types are annotated",
	},
	types: [SchemaFactory.string, SchemaFactory.number],
```
